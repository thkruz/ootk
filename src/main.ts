/**
 * @author @thkruz Theodore Kruczek
 * @description Orbital Object ToolKit (ootk) is a collection of tools for working
 * with satellites and other orbital objects.
 * @license AGPL-3.0-or-later
 * @copyright (c) 2020-2024 Theodore Kruczek
 *
 * Many of the classes are based off of the work of @david-rc-dayton and his
 * Pious Squid library (https://github.com/david-rc-dayton/pious_squid).
 *
 * Orbital Object ToolKit is free software: you can redistribute it and/or modify it under the
 * terms of the GNU Affero General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version.
 *
 * Orbital Object ToolKit is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License along with
 * Orbital Object ToolKit. If not, see <http://www.gnu.org/licenses/>.
 */

export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './time/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
export * from './operations/operations.js';
export { BaseObject, Sensor } from './objects/index.js';
export { Earth } from './body/index.js';
export * from './coordinate/index.js';
export * from './observation/index.js';
export * from './data/DataHandler.js';
export * from './sgp4/index.js';
export * from './objects/index.js';
export * from './interpolator/index.js';
export * from './force/index.js';
export * from './maneuver/index.js';
export * from './optimize/index.js';
export * from './orbit_determination/index.js';
export * from './body/index.js';
export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './coordinate/index.js';
export * from './operations/index.js';
export * from './force/index.js';
export * from './propagator/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
