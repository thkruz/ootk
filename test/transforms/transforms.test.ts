import { DEG2RAD, Degrees, Kilometers, Radians, RfSensor, SpaceObjectType, azel2uv, rae2raeOffBoresight, uv2azel } from '../../src/main.js';

// uv2azel
it('should convert valid unit vector to azimuth and elevation', () => {
  const u = 0 as Radians;
  const v = 0 as Radians;

  const azelCoordinates = uv2azel(u, v, (5 * DEG2RAD) as Radians);

  expect(azelCoordinates.az).toMatchSnapshot();
  expect(azelCoordinates.el).toMatchSnapshot();
});

// rae2raeOffBoresight
it('should convert valid RAE coordinates to RAE Off Boresight', () => {
  const rae = {
    rng: 0 as Kilometers,
    az: 0 as Degrees,
    el: 0 as Degrees,
  };

  const senor = new RfSensor({
    type: SpaceObjectType.PHASED_ARRAY_RADAR,
    lat: 0 as Degrees,
    lon: 0 as Degrees,
    alt: 0 as Kilometers,
    minAz: 0 as Degrees,
    maxAz: 0 as Degrees,
    minEl: 0 as Degrees,
    maxEl: 0 as Degrees,
    minRng: 0 as Kilometers,
    maxRng: 0 as Kilometers,
    boresightAz: [0 as Degrees],
    boresightEl: [0 as Degrees],
    beamwidth: 0 as Degrees,
  });

  const raeOffBoresightCoordinates = rae2raeOffBoresight(rae, senor, 0, 10 as Degrees);

  expect(raeOffBoresightCoordinates).toMatchSnapshot();
});

// eslint-disable-next-line multiline-comment-style
// rae2ruv
// it('should convert valid RAE coordinates to RUV', () => {
//   const rae = {
//     rng: 0 as Kilometers,
//     az: 0 as Degrees,
//     el: 0 as Degrees,
//   };
//   const ruvCoordinates = rae2ruv(rae, 0 as Degrees, 0 as Degrees, 0 as Degrees);
//   expect(ruvCoordinates).toMatchSnapshot();
// });

// azel2uv
it('should convert valid azimuth and elevation to unit vector', () => {
  const az = 0 as Radians;
  const el = 0 as Radians;

  const uvCoordinates = azel2uv(az, el, (5 * DEG2RAD) as Radians);

  expect(uvCoordinates.u).toMatchSnapshot();
  expect(uvCoordinates.v).toMatchSnapshot();
});
