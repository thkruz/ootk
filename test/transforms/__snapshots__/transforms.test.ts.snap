// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert valid RAE coordinates to RAE Off Boresight 1`] = `
{
  "az": 0,
  "el": 0,
}
`;

exports[`should convert valid azimuth and elevation to unit vector 1`] = `0`;

exports[`should convert valid azimuth and elevation to unit vector 2`] = `0`;

exports[`should convert valid unit vector to azimuth and elevation 1`] = `0`;

exports[`should convert valid unit vector to azimuth and elevation 2`] = `0`;
